lib/grandpa/message_handler.go

```diff
@@ -292,26 +309,42 @@ func (h *MessageHandler) verifyCommitMessageJustification(fm *CommitMessage) err
 }
 
 func (h *MessageHandler) verifyPreVoteJustification(msg *CatchUpResponse) (common.Hash, error) {
+	voters := make(map[ed25519.PublicKeyBytes]map[common.Hash]int)
+	eqVotesByHash := make(map[common.Hash]map[ed25519.PublicKeyBytes]struct{})
+
+	// identify equivocatory votes by hash
+	for _, just := range msg.PreVoteJustification {
+		voters[just.AuthorityID][just.Vote.Hash]++
+
+		if voters[just.AuthorityID][just.Vote.Hash] > 1 {
+			eqVotesByHash[just.Vote.Hash][just.AuthorityID] = struct{}{}
```

Comment:
You need to make the sub-map for `eqVotesByHash` if it's not existing (check with `_, ok := eqVotesByHash[just.Vote.Hash])